docket/plugins/config_scripts.py

```diff
@@ -0,0 +1,167 @@
+        attachment = msg.message.attachments[0]
+        content = str(await attachment.read())
+    else:
+        content = msg.message.content
+        content = msg.message.content.strip("```lua\n").strip("```")
+    return content
+
+
+async def script_name_autocomplete(
+    ctx: crescent.Context, option: hikari.AutocompleteInteractionOption
+) -> list[hikari.CommandChoice]:
+    if not ctx.guild_id:
+        return []
+    prefix = cast(str, option.value)
+    scripts = await Script.fetch_query().where(guild_id=ctx.guild_id).fetchmany()
+    script_names = [script.name for script in scripts]
+    return [
+        hikari.CommandChoice(name=name, value=name)
+        for name in get_close_matches(prefix, script_names, 10, 0.2)
+    ]
+
+
+@plugin.include
+@scripts.child
+@crescent.command(name="create", description="Create a new script")
+class CreateScript:
+    name = crescent.option(str, "The name of the script")
+
+    async def callback(self, ctx: crescent.Context) -> None:
+        name = self.name.lower().casefold().replace(" ", "-")
+        assert ctx.guild_id
+        await ctx.respond(
+            "Please send the Lua script (either in a code block or file " "upload)."
+        )
+        content = await wait_for_script(ctx)
+        if not content:
+            await ctx.edit("Script creation timed out.")
+
+        await Guild.get_or_create(ctx.guild_id)
+        try:
+            await Script(guild_id=ctx.guild_id, name=name, code=content).create()
+        except asyncpg.UniqueViolationError:
+            await ctx.edit("A script with that name already exists.")
+            return
+
+        await ctx.edit(f"Script '{name}' created.")
+
+
+@plugin.include
+@scripts.child
+@crescent.command(name="delete", description="Delete a script")
+class DeleteScript:
+    name = crescent.option(
+        str, "The name of the script", autocomplete=script_name_autocomplete
+    )
+
+    async def callback(self, ctx: crescent.Context) -> None:
+        assert ctx.guild_id
+        script = await Script.exists(name=self.name, guild_id=ctx.guild_id)
+        if not script:
+            raise ScriptNotFound(self.name)
+        await script.delete()
+        await ctx.respond(f"Script '{self.name}' deleted.")
+
+
+@plugin.include
+@scripts.child
+@crescent.command(name="edit", description="Edit a script")
+class EditScript:
+    name = crescent.option(
+        str, "The name of the script", autocomplete=script_name_autocomplete
+    )
+
+    async def callback(self, ctx: crescent.Context) -> None:
+        assert ctx.guild_id
+        script = await Script.exists(name=self.name, guild_id=ctx.guild_id)
+        if not script:
+            raise ScriptNotFound(self.name)
+        await ctx.respond(
+            "Please send the Lua script (either in a code block or file " "upload)."
+        )
+        content = await wait_for_script(ctx)
+        if not content:
+            await ctx.edit("Script creation timed out.")
+            return
```

Comment:
I think this can also be handled with a custom exception and `crescent.catch_command`

Reply:
Feels like that over complicates stuff, and makes it harder to understand. This is sorta an edge case anyways.

Reply:
I think my earlier comment is for the exact same usecase

Reply:
Already went over this in a different comment so I'm resolving